tests/pool_alloc.c

```diff
@@ -0,0 +1,70 @@
+#include <pool_alloc.h>
+#include <stdc.h>
+#include <malloc.h>
+#include <ktest.h>
+
+void int_constr(void *buf, __attribute__((unused)) size_t size) {
+  int *num = buf;
+  *num = 0;
+}
+void int_destr(__attribute__((unused)) void *buf,
+               __attribute__((unused)) size_t size) {
```

Comment:
I have a feeling it will be common for many simple objects allocated with the pool allocator to have no destructor. Well, maybe. In that case I would need to provide such pointless destructor function just to satisfy the interface. It might be convenient to have a no-op `pool_default_destructor` defined in `pool_alloc.c` so that one might:
```c
pool_init(&my_pool, size, my_constr, pool_default_destructor);
```
without creating additional empty destructor. We do a similar thing with `vnode` operations: a number of trivial `vnode_generic_*` is provided so that one does not have to write their own empty functions.